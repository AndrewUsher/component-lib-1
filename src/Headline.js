import React from 'react'
import styled, { css } from 'styled-components'
import { hideOverflowStyles } from './utils'

const BaseHeadline = styled.div`
  font-family: helvetica, arial, sans-serif;
  letter-spacing: -0.025em;
  color: #202124;
  margin: 0;
  line-height: 1;
  ${props => props.hideOverflow && hideOverflowStyles}
`

const StyledHeadline1 = styled(BaseHeadline)`
  font-size: 3.5rem;
  font-weight: 600;
`

const StyledHeadline2 = styled(BaseHeadline)`
  font-size: 3rem;
  font-weight: 500;
`

const StyledHeadline3 = styled(BaseHeadline)`
  font-size: 2.5rem;
  font-weight: 500;
`

const Headline1 = props => <StyledHeadline1 as="h1" {...props} />
const Headline2 = props => <StyledHeadline2 as="h2" {...props} />
const Headline3 = props => <StyledHeadline3 as="h3" {...props} />

const levelMap = { 1: Headline1, 2: Headline2, 3: Headline3 }

const Headline = ({ hideOverflow, level, ...props }) => {
  const Component = levelMap[level]
  return <Component hideOverflow={hideOverflow} {...props} />
}

Headline.defaultProps = {
  hideOverflow: false,
  level: 1
}

export default Headline
