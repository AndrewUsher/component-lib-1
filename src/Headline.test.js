import React from 'react'
import { cleanup, render } from 'react-testing-library'
import Headline from './Headline'
import 'jest-styled-components'

describe('<Headline />', () => {
  afterEach(cleanup)

  test('Renders level 1 correctly', function level1Test() {
    const { container, getByText } = render(
      <Headline level="1">Hello</Headline>
    )
    expect(getByText('Hello')).toBeTruthy()
    expect(container).toMatchSnapshot()
  })
})
