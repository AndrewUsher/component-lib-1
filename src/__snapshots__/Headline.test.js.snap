// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Headline /> Renders level 1 correctly 1`] = `
.c0 {
  font-family: helvetica,arial,sans-serif;
  -webkit-letter-spacing: -0.025em;
  -moz-letter-spacing: -0.025em;
  -ms-letter-spacing: -0.025em;
  letter-spacing: -0.025em;
  color: #202124;
  line-height: 1;
  font-size: 3.5rem;
  font-weight: 600;
}

<div>
  <h1
    class="c0"
  >
    Hello
  </h1>
</div>
`;
